Apple SSL

### Describe the bug

Assertion failed in registering to stun server

`16:01:07.080            pjsua_acc.c  ..Acc 0: Registration sent
registering...
16:01:07.145            stunresolve  .STUN mapped address found/changed: XX.XXX.11.166:19158
16:01:07.146           pjsua_core.c  .STUN resolution success, using 193.140.74.144:3478, address is <STUN_IP>:<STUN_PORT>
Assertion failed: (ssock->send_buf.max_len == 0), function ssock_on_connect_complete, file ../src/pj/ssl_sock_imp_common.c, line 1134.`




### Steps to reproduce

```
void PJSua2::createLib() {
    try {
        ep->libCreate();
    } catch (Error& err){
        std::cout << "Startup error: " << err.info() << std::endl;
    }
    
    //LibInit
    try {
        ep->libInit( *ep_cfg );
    } catch(Error& err) {
        std::cout << "Initialization error: " << err.info() << std::endl;
    }
    
    // Create SIP transport. Error handling sample is shown
    try {
    TransportConfig tcfg;
    
    //TLS
    pjsip_cfg();
    tcfg.tlsConfig.method = PJSIP_TLSV1_3_METHOD;
    tcfg.port = 0; //-> Check w/zero
    ep->transportCreate(PJSIP_TRANSPORT_TLS, tcfg);
    ep->transportCreate(PJSIP_TRANSPORT_UDP, tcfg);

    } catch(Error& err) {
    std::cout << "Transport creation error:" << err.info() << std::endl;
    }

    // Start the library (worker threads etc)
    try { ep->libStart();
    } catch(Error& err) {
    std::cout << "Startup error: " << err.info() << std::endl;
    }
}

```

Then create acc:

```
void PJSua2::createAccount(std::string username, std::string password, std::string ip, std::string port, std::string stun_ip, std::string stun_port, std::string turn_ip, std::string turn_port, std::string turn_username, std::string turn_password, bool ice_enabled, bool turn_enabled, bool tls_enabled, bool stun_enabled) {
    
    
    // Configure an AccountConfig
    AccountConfig acfg;

    ep_cfg->uaConfig.userAgent = "IPHONE";

    
    //NAT CONFIGs
    acfg.natConfig.iceEnabled = PJ_TRUE;
    
    //Set Turn
    if(turn_enabled){
        std::string turnIp = turn_ip;
        std::string turnUsername = turn_username;
        std::string turnPassword = turn_password;
        std::string turnPort = turn_port;
    }
    
    //Set Stun
    if(stun_enabled){
        ep_cfg->uaConfig.stunServer.push_back(stun_ip + ":" + stun_port);
    }
    
    //TLS CONFIG
    if(tls_enabled){
        acfg.sipConfig.proxies.clear();
        acfg.sipConfig.proxies.push_back("sip:" + ip + ";hide;transport=tls");
    }
    
    // Ice Configs
    if(ice_enabled){
        
        
        //ICE
        acfg.natConfig.iceMaxHostCands = -1;
        
        //STUN
        acfg.natConfig.sipStunUse = PJSUA_STUN_USE_DEFAULT;
        acfg.natConfig.mediaStunUse = PJSUA_STUN_RETRY_ON_FAILURE;
        ep->natUpdateStunServers(ep_cfg->uaConfig.stunServer, PJ_FALSE);
        
        //TURN
        acfg.natConfig.turnEnabled = PJ_TRUE;
        acfg.natConfig.turnServer = turn_ip + ":" + turn_port;
        acfg.natConfig.turnUserName = turn_username;
        acfg.natConfig.turnPassword = turn_password;
    }
    
    else if (turn_enabled){
        acfg.natConfig.sipStunUse = PJSUA_STUN_USE_DISABLED;
        acfg.natConfig.mediaStunUse = PJSUA_STUN_USE_DISABLED;
        acfg.natConfig.iceMaxHostCands = 0; 
        acfg.natConfig.turnEnabled = PJ_TRUE;
        acfg.natConfig.turnServer = turn_ip+":"+turn_port;
        acfg.natConfig.turnUserName = turn_username;
        acfg.natConfig.turnPassword = turn_password;
    }
    else if (stun_enabled){
        acfg.natConfig.sipStunUse = PJSUA_STUN_USE_DEFAULT;
        acfg.natConfig.mediaStunUse = PJSUA_STUN_RETRY_ON_FAILURE;
        acfg.natConfig.iceMaxHostCands = -1;
        
        //bj_bzero -> std::fill
        std::fill(EpConfig().uaConfig.stunServer.begin(), EpConfig().uaConfig.stunServer.end(), 0);
        ep->natUpdateStunServers(EpConfig().uaConfig.stunServer, PJ_FALSE);
    
    } else {
        acfg.natConfig.iceEnabled = PJ_FALSE;
        acfg.natConfig.turnEnabled = PJ_FALSE;
        acfg.natConfig.sipStunUse = PJSUA_STUN_USE_DISABLED;
        acfg.natConfig.mediaStunUse = PJSUA_STUN_USE_DISABLED;
    }
    
    // Configure an AccountConfig
    acfg.idUri = "sip:" + username + "@" + ip + ":" + port;
    acfg.regConfig.registrarUri = "sip:" + ip + ":" + port;
    AuthCredInfo cred("digest", "*", username, 0, password);
    acfg.sipConfig.authCreds.push_back(cred);


    acfg.sipConfig.contactUriParams=";app-id="+appId+";pn-type=1"+";pn-tok="+apnToken; //PUSH NOTIFICATION SUPPORT //according to linphone standart
    
    //acfg.regConfig.registrarUri = "sip:" + ip + ":" + port + ";transport=tcp"; // TCP SUPPORT
    

    if(acc->getId() == -1){
        // Create the account
        try {
            acc->create(acfg);
        } catch(Error& err) {
            std::cout << "Account creation error: " << err.info() << std::endl;
        }
    }else {
        // Modify the account
        try {
            //Update the registration
            acc->modify(acfg);
            acc->setRegistration(true);
        } catch(Error& err) {
            std::cout << "Account modify error:  " << err.info() << std::endl;
        }
    }
}
```


### PJSIP version

2.11

### Context

The issue happens iphone 12 w/configure-iphone parameter w/third-party-media-stack and i'm using APPLE SSL

### Log, call stack, etc

```shell
pjsua_core.c  Trying STUN server <IP:PORT> IPv4 (1 of 1)..
16:01:07.077            pjsua_acc.c  Adding account: id=sip:<SIPID>@<IP:PORT>
16:01:07.077            pjsua_acc.c  .Account sip:ID@<IP:PORT> added with id 0
16:01:07.077            pjsua_acc.c  .Acc 0: setting registration..
16:01:07.079        tlsc0x10a817228  ..TLS client transport created
16:01:07.079        tlsc0x10a817228  ..TLS transport <IP:PORT> is connecting to <IP:PORT>
16:01:07.079            pjsua_acc.c  ..Contact for acc 0 updated: <sip:<IP:PORT>;transport=TLS;app-id=;pn-type=1;ob>;+sip.ice;reg-id=1;+sip.instance="<urn:uuid:00000000-0000-0000-0000-00008340d0fa>"
16:01:07.080           pjsua_core.c  ...TX 743 bytes Request msg REGISTER/cseq=50721 (tdta0x10a81a8a8) to TLS <IP:PORT>
REGISTER sip:<IP:PORT> SIP/2.0

Via: SIP/2.0/TLS <IP:PORT>;rport;branch=;alias

Max-Forwards: 70

From: <sip:@>;tag=1saYM2e3YOCvnWPxKGfRHKswBKN8zURJ

To: <sip:@>

Call-ID:

CSeq: 50721 REGISTER

Supported: outbound, path

Contact: <sip:<id>@<IP:PORT>;transport=TLS;pn-type=1;pn-tok=;ob>;+sip.ice;reg-id=1;+sip.instance="<urn:uuid:00000000-0000-0000-0000-00008340d0fa>"

Expires: 300

Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS

Content-Length:  0




--end msg--
16:01:07.080            pjsua_acc.c  ..Acc 0: Registration sent
registering...
16:01:07.145            stunresolve  .STUN mapped address found/changed: <IP:PORT>
16:01:07.146           pjsua_core.c  .STUN resolution success, using <IP:PORT>, address is <IP:PORT>
Assertion failed: (ssock->send_buf.max_len == 0), function ssock_on_connect_complete, file ../src/pj/ssl_sock_imp_common.c, line 1134.
```

**DICTIONARY:**
TP: THIRD PARTY
SP: SOFT PHONE (DEFAULT APPS) (pjsua2 app is same with the waiting pr)
OPENSSL VER: 1.1.1M
LAST UPDATE PJSIP VER: 09/03/2022 09:10 (GMT +3)
(+) -> BUILD SUCCESS
(-) -> BUILD FAILED


IPHONE SE2 - IOS 15.3.1 RESULTS:
|         | AS                               | AS                                        | AS                               | OS                               | OS                                        | OS                                 |   |
|---------|----------------------------------|-------------------------------------------|----------------------------------|----------------------------------|-------------------------------------------|------------------------------------|---|
|         | PJSUA2-SWIFT                     | PJSUA-SWIFT                               | IPJSUA                           | PJSUA2-SWIFT                     | PJSUA-SWIFT                               | IPJSUA                             |   |
| TP-LTE  |                                  |                                           |                                  |                                  |                                           |                                    |   |
| TP-WIFI |                                  |                                           |                                  |                                  |                                           |                                    |   |
| SP-LTE  | (+) https://file.io/RsdpBV48t57a | Undefined symbol: _pjsua_vid_win_get_info | (+) https://easyupload.io/2wtpi1 | (+) https://file.io/OVgd8EzYkxOf | Undefined symbol: _pjsua_vid_win_get_info | (+)   https://file.io/GOq8WQnHvJfb |   |
| SP-WIFI | (+) https://file.io/gebnclTRcpGm | Undefined symbol: _pjsua_vid_win_get_info | (+) https://easyupload.io/9blvaw | (+) https://file.io/2245ApQS18RA | Undefined symbol: _pjsua_vid_win_get_info | (+)   https://file.io/AARR9rmH4AyJ |   |
|         |                                  |                                           |                                  |                                  |                                           |                                    |   |


IPHONE 12 - IOS 14.6:

|         | AS                               | AS                                        | AS                                                              | OS                               | OS                                        | OS                               |   |
|---------|----------------------------------|-------------------------------------------|-----------------------------------------------------------------|----------------------------------|-------------------------------------------|----------------------------------|---|
|         | PJSUA2-SWIFT                     | PJSUA-SWIFT                               | IPJSUA                                                          | PJSUA2-SWIFT                     | PJSUA-SWIFT                               | IPJSUA                           |   |
| TP-LTE  |                                  |                                           |                                                                 |                                  |                                           |                                  |   |
| TP-WIFI |                                  |                                           |                                                                 |                                  |                                           |                                  |   |
| SP-LTE  | (-) https://file.io/gdieEadhMUvW | Undefined symbol: _pjsua_vid_win_get_info | (-)https://file.io/revNL8PDVDj3                                 | (+) https://file.io/yPPkwsOhPkAN | Undefined symbol: _pjsua_vid_win_get_info | (+) https://file.io/NJgDHf4VoEkN |   |
| SP-WIFI | (+) https://file.io/kIp41DNV2Rrv | Undefined symbol: _pjsua_vid_win_get_info | (+)https://file.io/7JXc1NC5otwh (-)https://file.io/Z0mbtHgfjNcT | (+) https://file.io/LVVQLH389hUe | Undefined symbol: _pjsua_vid_win_get_info | (+) https://file.io/OgNekbimOsst |   |
|         |                                  |                                           |                                                                 |                                  |                                           |                                  |   |

--- a/pjlib/src/pj/ssl_sock_apple.m
+++ b/pjlib/src/pj/ssl_sock_apple.m
@@ -675,6 +675,9 @@ static pj_status_t verify_cert(applessl_sock_t *assock, pj_ssl_cert_t *cert)
         
         err = SecTrustGetTrustResult(trust, &trust_result);
         if (err == noErr) {
+#if SSL_DEBUG
+            PJ_LOG(3, (THIS_FILE, "SSL trust evaluation: %d", trust_result));
+#endif
     	    switch (trust_result) {
     		case kSecTrustResultInvalid:
 		    ssock->verify_status |= PJ_SSL_CERT_EINVALID_FORMAT;
@@ -1027,7 +1030,7 @@ static pj_status_t network_create_params(pj_ssl_sock_t * ssock,
 	      sec_protocol_metadata_get_negotiated_tls_ciphersuite(metadata);
 
 	    /* For client, call on_connect_complete() callback first. */
-	    if (!ssock->is_server) {
+	    if (!ssock->is_server && ssock->ssl_state == SSL_STATE_NULL) {
 	    	if (!assock->connection)
 	    	    complete(false);
 

